Dockerfile

```diff
@@ -15,21 +15,25 @@
 ARG CONTAINERD_VERSION=1.6.30
 ARG RUNC_VERSION=1.1.12
 ARG NERDCTL_VERSION=1.7.1
-ARG GO_VERSION=1.21.9
-ARG REGISTRY_VERSION=3.0.0-alpha.1
 
-FROM public.ecr.aws/docker/library/golang:${GO_VERSION}-bookworm AS golang-base
+FROM public.ecr.aws/docker/library/golang:1.21.9-alpine AS containerd-snapshotter-base
 
-FROM golang-base AS containerd-snapshotter-base
 ARG CONTAINERD_VERSION
 ARG RUNC_VERSION
 ARG NERDCTL_VERSION
-ARG GO_VERSION
 ARG TARGETARCH
 COPY ./integ_entrypoint.sh /integ_entrypoint.sh
 COPY . $GOPATH/src/github.com/awslabs/soci-snapshotter
 ENV GOPROXY direct
-RUN apt-get update -y && apt-get install -y libbtrfs-dev libseccomp-dev libz-dev gcc fuse pigz
+RUN apk add --no-cache \
+    btrfs-progs-libs \
+    curl \
+    fuse \
+    gcc \
+    libc6-compat \
```

Comment:
Also considered using statically linked binaries or building from source (for containerd, nerdctl, and runc) instead of libc6-compat.